hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmComparisonPredicate.java

```diff
@@ -41,6 +45,12 @@ private SqmComparisonPredicate(
 		this.rightHandExpression = rightHandExpression;
 		this.operator = operator;
 
+		if ( leftHandExpression instanceof SqmPluralValuedSimplePath ) {
+			throw new SemanticException( "The equal operator cannot be applied to a multi-valued node" );
```

Comment:
I guess you introduced this because `assertComparable` does not fail right now for such plural valued paths? If so, please try to move this check somewhere into `assertComparable`.